tools/run_tests/helper_scripts/build_python.sh

```diff
@@ -142,6 +142,11 @@ fi
  true)
 VENV_PYTHON=$(script_realpath "$VENV/$VENV_RELATIVE_PYTHON")
 
+# See https://github.com/grpc/grpc/issues/14815 for more context. We cannot rely
+# on pip to upgrade itself because if pip is too old, it may not have the required
+# TLS version to run `pip install`.
+curl https://bootstrap.pypa.io/get-pip.py | $VENV_PYTHON
```

Comment:
My understanding was that ```pip``` was using the python ```ssl``` module provided by the Python distribution.

If the TLS version is too old to use pip to upgrade pip here, how does it work down below when we call ```$VENV_PYTHON -m pip install --upgrade pip==9.0.2``` ?

Reply:
I think `pip <9.0.2` tries to use OpenSSL for TLS, which is outdated on macOS because Apple decided to SecureTransport for TLS. In `pip 9.0.2`, the fallback to SecureTransport is added.

`$VENV_PYTHON -m pip install --upgrade pip==9.0.2` doesn't work (anymore). The old workers had brew-managed Python 3.4, but now the brew-managed versions on Sierra are 2.7 and 3.6. So this script now fails when using Python 3.4.